process/setup_sp.py

```diff
@@ -32,7 +33,7 @@ def read_proj_graphml(proj_graphml_filepath, ori_graphml_filepath, to_crs,undire
         make graph undirected
     retain_edge_attributes = list (default: None)
         explicitly retain only a subset of edge attributes, else keep all (default)
-     
+

```

Comment:
I'm noticing that probably 1/2 of the diffs here are whitespace changes. I wonder if we should enforce code guidelines and formatting (like black) to keep things more consistent.

Reply:
I think that's a good idea.  Regarding code guidelines and formatting, we could try using a code linter to guide style enforcement perhaps @shiqin-liu @gboeing ?  I haven't used one before but I've been meaning to start trying to make a habit of it; sounds like it might be appropriate for this issue.  This [blog post]( https://dbader.org/blog/python-code-linting ) recommends Flake8 for conformity to PEP8 style. Maybe either that or PyLint could be useful ([reddit thread here](https://www.reddit.com/r/Python/comments/82hgzm/any_advantages_of_flake8_over_pylint/))?  Would you have any recommendations or thoughts around this Shirley or Geoff?

Reply:
I'd recommend a setup similar to that of OSMnx: isort + black + flake8 + pydocstyle for linting. And then build/unit tests as needed. OSMnx linting and testing is [here](https://github.com/gboeing/osmnx/tree/master/tests) and some [configs](https://github.com/gboeing/osmnx/blob/master/setup.cfg) are by necessity at the project root.

Reply:
One other comment: this all uses continuous integration, which is configured [here](https://github.com/gboeing/osmnx/blob/master/.github/workflows/tests.yml) as an example.

Reply:
I like the sounds of it -- although I wouldn't be sure how to get started with it all; I'll try to look into this later in the week so I'm familiar with the process at least; maybe we can talk about possible implementation in our meeting later in the week?

Reply:
Thanks Geoff. I noticed the space change but had no idea why this happened... It sounds a great idea to enforce code guidelines and formatting to deal with this. This is quite new to me, I will aim to take a look at the examples and guides. I would appreciate if someone could lend a hand too :)

Reply:
I can take the lead getting continuous integration and linting set up.